### DISCUSSION TITLE
SMTP authentication with OAuth in Office 365

### DISCUSSION THREAD

**Discussion by caupoab** [pimcore/pimcore#13456] [Q&A / Support]
Microsoft is moving away from basic authentication for email protocols. One of our customers experienced an issue where Pimcore could not authenticate with Office 365 since they changed something related to the authentication. Since then the customer seems to have reenabled the STMP authentication in Office 365. But we are still worried since we suspect this might only be a temroray fix if Microsoft will only allow OAuth in a not too distant future.

 In [this post from Microsoft](https://techcommunity.microsoft.com/t5/exchange-team-blog/basic-authentication-deprecation-in-exchange-online-september/ba-p/3609437) it says SMTP auth is not in  scope for this disablement. It's not crystal clear to me if and when they plan to remove SMTP auth.

Is there a way to use OAuth for email authentication with Pimcore? Microsoft has [a guide about it here](https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth).

If this is not possible in Pimcore at the moment, is it a feature you are working on or are planning to implement?

Best regards,
Olof Johansson
Caupo

**Comment by studio1gmbh-pimcore** [comment]
We are also interested to know to what extent it is possible / what is needed to be able to send emails from Pimcore using OAuth.
Many thanks and best regards.

**Comment by Bhawana483** [comment]
@caupoab Did you get any solution?

**Comment by thomasschradermasnou** [comment]
Send emails with Symfony Mailer through Outlook / office365 with OAuth:
https://gist.github.com/dbu/3094d7569aebfc94788b164bd7e59acc

**Comment by thomasschradermasnou** [comment]
Or 
Symfony Mailer Transport for Microsoft Office 365 Graph API:

https://github.com/vitrus/symfony-office-graph-mailer/tree/master
